content/git/workflow_rstudio.md

```diff
@@ -0,0 +1,106 @@
+---
+date: 2017-10-18T14:42:43+02:00
+description: "Git collaborative workflow for Rstudio"
+title: Workflow Rstudio
+---
+
+# BEFORE I START WORKING
+
+## STEP 1: 
+
+**Choose your option:**
+
+### OPTION 1A: *I'll make a new branch to work with*  
+Create a new topic branch from master
+
+Make sure you are working on the master branch:
+![](../images/workflow_rstudio_1.png)
+
+Update the master branch to make sure it is aligned with the remote master: Click `Pull`:
+![](../images/workflow_rstudio_2.png)
+
+Create a new branch: click on `New branch` and fill in a branch name:
+![](../images/workflow_rstudio_2b.png)
+
+### **OPTION 1B**:  *I already have a branch I want to continue working on*:  
+
+Switch to existing topic branch:
+![](../images/workflow_rstudio_3.png)
+
+Update the topic branch to make sure is is aligned with the remote master: 
+In the Git shell, type:
+```
+git pull origin master
+```
+![](../images/workflow_rstudio_4.png)
+
+
+# WHILE EDITING
+
+## STEP 2.x: adapt in md, R-code,... (multiple times!)
+
+### Add new files
+
+Stage the new file with checkbox and click `Commit`
+![](../images/workflow_rstudio_5.png)
+
+Provide clear and understandable message about adaptations
+![](../images/workflow_rstudio_6.png)
+
+### Change files:
+
+Stage the adaptations with checkbox and click `Commit`
+![](../images/workflow_rstudio_7.png)
+
+Provide clear and understandable message about adaptations
+![](../images/workflow_rstudio_8.png)
+
+
+## STEP 3: Push your changes to GitHub.
+
+Update the remote version of the topic branch. Click `Push`:
+![](../images/workflow_rstudio_9.png) 
+
+# EDITS ON BRANCH ARE READY
+
+## STEP 4: Pull request to add your changes to the current master. 
+
+Update the remote version of the topic branch. Click `Push`:
+![](../images/workflow_rstudio_9.png)
+
+Go to your repo on Github.com and click the `Compare & pull request` button. 
+
+![](../images/workflow_rstudio_github_1.png)
+
+Add information about the Pull request, add reviewerrs, labels,... Finally, `Create pull request`:
+![](../images/workflow_rstudio_github_2.png)
+
+## STEP 5: Code review!
+
+You and collaborators can make comments about the edits, review the code and adapt if required (create additional commits and `Push` them to GitHub) .
```

Comment:
I think we should suggest some rules for collaboration. E.g. https://www.muscardinus.be/2017/08/git_workflow_ecologists/

Reply:
Maybe out of scope for this document. But IMHO very important to teach right for the start. Otherwise people start to cooperate on a repo an run into troubles which were easy to avoid if they would apply some basic rules. If you don't teach such rules, you can't blame the users that you didn't obeyed the rules.

Reply:
You're right about that. But I do believe that by this step by step workflow we actually can achieve the `Branching workflow with pull requests` from you're blogpost... I've been testing similar step by step workflows successfully with student-groups in the past.

Furthermore, i'll add the rules for collaboration from you're blog post:
- Always commit into a feature branch, never in the master branch.
- Always start features branches for the master branch.
- Only work in your own branches.
- Never merge someone else’s pull request without their consent.
- Don’t wait too long for merging a branch. Keep the scope of a feature branch narrow.

Into the course itself.

Reply:
https://github.com/inbo/git-course/commit/43ea1226a33d7d8e6dcddf9fd7f510acdd128446